src/pas/plugins/oidc/plugins.py

```diff
@@ -331,8 +362,19 @@ def _setupJWTTicket(self, user_id, user):
 
     # TODO: memoize (?)
     def get_oauth2_client(self):
+        domain = self.getProperty("identity_domain_name")
         try:
-            client = Client(client_authn_method=CLIENT_AUTHN_METHOD)
+            if domain:
+                client = OAMClient(
+                    client_authn_method=CLIENT_AUTHN_METHOD,
+                    domain=domain,
+                )
+            else:
+                client = Client(client_authn_method=CLIENT_AUTHN_METHOD)
+            client.allow["issuer_mismatch"] = (
```

Comment:
Yes! I noticed the same issue with the Microsoft implementation. However, in my opinion, it might be better to have this in a separate property, even though this plugin already seems to have too many configurations.

Reply:
I can put a toggle for it. It's not exactly clear what the implications are if allow mismatch. I guess its less secure? Could have a toggle that defaults mismatch allowed so it's less likely to trip up first timers?